src/gas_metering/mod.rs

```diff
@@ -141,27 +145,33 @@ pub fn inject<R: Rules>(
 	gas_module_name: &str,
 ) -> Result<elements::Module, elements::Module> {
 	// Injecting gas counting external
-	let mut mbuilder = builder::from_module(module);
-	let import_sig =
-		mbuilder.push_signature(builder::signature().with_param(ValueType::I32).build_sig());
 
+	let mut mbuilder = builder::from_module(module);
 	mbuilder.push_import(
 		builder::import()
 			.module(gas_module_name)
 			.field("gas")
```

Comment:
Can we call this `gas_counter`, or something like that, and store it in the "gas" module?